src/mame/audio/cage.cpp

```diff
@@ -441,37 +441,39 @@ void atari_cage_device::update_control_lines()
 	/* set the IRQ to the main CPU */
 	int reason = 0;
 
-	if ((m_control & 3) == 3 && !m_cpu_to_cage_ready)
+	if ((m_control & 3) == 3 && !(m_soundlatch[1]->pending_r()))
 		reason |= CAGE_IRQ_REASON_BUFFER_EMPTY;
-	if ((m_control & 2) && m_cage_to_cpu_ready)
+	if ((m_control & 2) && m_soundlatch[0]->pending_r())
 		reason |= CAGE_IRQ_REASON_DATA_READY;
 
 	m_irqhandler(machine().dummy_space(), 0, reason);
 	/* set the IOF input lines */
 	val = m_cpu->state_int(TMS3203X_IOF);
 	val &= ~0x88;
-	if (m_cpu_to_cage_ready) val |= 0x08;
-	if (m_cage_to_cpu_ready) val |= 0x80;
+	if (m_soundlatch[1]->pending_r()) val |= 0x08;
+	if (m_soundlatch[0]->pending_r()) val |= 0x80;
 	m_cpu->set_state_int(TMS3203X_IOF, val);
 }
 
 
 READ32_MEMBER( atari_cage_device::cage_from_main_r )
 {
 	if (LOG_COMM)
-		logerror("%s CAGE read command = %04X\n", machine().describe_context(), m_from_main);
-	m_cpu_to_cage_ready = 0;
+		logerror("%s CAGE read command = %04X\n", machine().describe_context(), m_soundlatch[1]->read(space, 0));
+
 	update_control_lines();
-	m_cpu->set_input_line(TMS3203X_IRQ0, CLEAR_LINE);
-	return m_from_main;
+	if (LOG_COMM)
+		m_soundlatch[1]->acknowledge_w(space, 0, 0);
```

Comment:
This can't possibly be right - changing logging settings shouldn't affect behaviour.  Read the value and store it in a local variable before the `logerror` or something.